### PR TITLE
Rani 01 patch 1

### PR DESCRIPTION
**Pull request by Rani-01**
updated database with sqlalchemy and sqllite3
used class objects for student details
removed assert function.

First install packages sqlalchemy and unittest using commands
pip install sqlalchemy
pip install unittest
in terminal
first run school_management.py and then test.py using commands
python school_management.py
python test.py

### CODE REVIEW

File: school_management.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+    try:
+        cursor.execute('UPDATE SCHOOL_MANAGEMENT SET NAME=?, EMAIL=?, PHONE_NO=?, GENDER=?, DOB=?, STREAM=? WHERE STUDENT_ID=?',
+                       (new_data['name'], new_data['email'], new_data['phone_no'], new_data['gender'], new_data['dob'], new_data['stream'], admission_no))
+        connector.commit()
+    except sqlite3.Error as e:
+        connector.rollback()
+        raise e
+    finally:
+        connector.close()
```
Comment (rkpagadala) [inline school_management.py:65]: remove

File: school_management.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        self.email = new_data['email']
+        self.phone_no = new_data['phone_no']
+        self.gender = new_data['gender']
+        self.dob = new_data['dob']
+        self.stream = new_data['stream']
+
+# Other functions and configurations here
+
+# Define the update_record function
+def update_record(admission_no, new_data):
```
Comment (rkpagadala) [inline school_management.py:43]: add as staticmethod to student class

File: test.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
+# test_school_management.py
+
+from datetime import datetime
+import unittest
+import school_management
+
+class TestUpdateRecord(unittest.TestCase):
+
+    def setUp(self):
```
Comment (rkpagadala) [inline test.py:9]: create test data and connect

File: test.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+import school_management
+
+class TestUpdateRecord(unittest.TestCase):
+
+    def setUp(self):
+        # Create the database and tables
+        Base.metadata.create_all(engine)
+        self.session = Session()
+
+    def tearDown(self):
```
Comment (rkpagadala) [inline test.py:14]: delete test data

File: test.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        # Clean up after the test
+        self.session.close()
+        Base.metadata.drop_all(engine)
```
Comment (rkpagadala) [inline test.py:17]: no drop all

File: test.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+        #Create an expected student object
+        expected_updated_student = Student(admission_no=admission_no, **updated_data)
+
+        # Check if the updated record values are the same
+        are_equal = self.are_students_equal(updated_student, expected_updated_student)
+
+        # You can now use 'are_equal' as a boolean to check if both objects are the same
+        return are_equal
+if __name__ == '__main__':
```
Comment (rkpagadala) [inline test.py:57]: new line

**Review by rkpagadala — CHANGES_REQUESTED** [review]

File: school_management.py (hunk 1 above)
Comment (Rani-01) [reply school_management.py:65]: removed

File: test.py (hunk 6 above)
Comment (Rani-01) [reply test.py:57]: done

File: test.py (hunk 5 above)
Comment (Rani-01) [reply test.py:17]: removed drop all

**Pull request merged**